packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -261,6 +279,7 @@ export class Pipeline extends PipelineBase {
           // remove the key - there is a grace period of a few days before it's gone for good,
           // that should be enough for any emergency access to the bucket artifacts
           removalPolicy: RemovalPolicy.DESTROY,
+          enableKeyRotation: this.enableKeyRotation,
```

Comment:
Let's pass `props.enableKeyRotation` here instead. This will save you from changing the existing templates when `enableKeyRotation` is not set.